src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveEditorContextTracker.cs

```diff
@@ -56,57 +52,42 @@ public int GetProjectName(uint itemid, out string pbstrProjectName)
             return HResult.OK;
         }
 
-        public bool IsActiveEditorContext(IWorkspaceProjectContext context)
+        public bool IsActiveEditorContext(string contextId)
         {
-            Requires.NotNull(context, nameof(context));
+            Requires.NotNullOrEmpty(contextId, nameof(contextId));
 
-            if (_contexts.TryGetValue(context, out string projectContextId))
-            {
-                return StringComparers.WorkspaceProjectContextIds.Equals(ActiveIntellisenseProjectContext, projectContextId);
-            }
+            if (!_contexts.Contains(contextId))
+                throw new InvalidOperationException($"'nameof{contextId}' has not been registered or has already been unregistered");
 
-            throw new InvalidOperationException("'context' has not been registered or has already been unregistered");
+            return StringComparers.WorkspaceProjectContextIds.Equals(ActiveIntellisenseProjectContext, contextId);
         }
 
-        public void RegisterContext(IWorkspaceProjectContext context, string contextId)
+        public void RegisterContext(string contextId)
         {
-            Requires.NotNull(context, nameof(context));
             Requires.NotNullOrEmpty(contextId, nameof(contextId));
 
-            bool changed = ThreadingTools.ApplyChangeOptimistically(ref _contexts,
-                                                                    projectContexts => projectContexts.SetItem(context, contextId));
+            bool changed = ThreadingTools.ApplyChangeOptimistically(ref _contexts, projectContexts =>
+            {
+                if (!projectContexts.Contains(contextId))
+                {
+                    projectContexts = projectContexts.Add(contextId);
+                }
+
+                return projectContexts;
+            });
 
             if (!changed)
-                throw new InvalidOperationException("'context' has already been registered.");
+                throw new InvalidOperationException($"'{nameof(contextId)}' has already been registered.");
```

Comment:
Based on this, looks like for my above comment nameof should be inside the braces, but perhaps the value of contextId help diagnosis too?